src/js_native_api_v8.h

```diff
@@ -0,0 +1,201 @@
+#ifndef SRC_JS_NATIVE_API_V8_H_
+#define SRC_JS_NATIVE_API_V8_H_
+
+#include <string.h>
+#include "js_native_api_v8_internals.h"
+
+struct napi_env__ {
+  explicit napi_env__(v8::Local<v8::Context> context)
+      : isolate(context->GetIsolate()),
+        context_persistent(isolate, context) {
+    CHECK_EQ(isolate, context->GetIsolate());
+  }
+  v8::Isolate* const isolate;  // Shortcut for context()->GetIsolate()
+  v8impl::Persistent<v8::Context> context_persistent;
+
+  inline v8::Local<v8::Context> context() const {
+    return StrongPersistentToLocal(context_persistent);
```

Comment:
Should this be v8impl::StrongPersistentToLocal like v8impl::Persistent?

Reply:
I spoke to @addaleax and as a result I'm going to make `StrongPersistentToLocal()` and `PersistentToLocal()` instance methods of `node::Persistent`, so we're gonna bring them in as part of the existing alias.

Reply:
... or rather as a new class `PersistentToLocal` with three static methods, because `node::Persistent` is itself an alias to `v8::Persistent` with the trait whereby no `Reset()` in the destructor is needed.